### PR TITLE
reduce the val 2022 samples with min gap of 1 hour

### PR DESCRIPTION
**Pull request by dfulu**
Reduce the 2022 validation samples to have a minimum gap between consecutive samples of 1 hour

I will run persistence model with and without this change as a sanity check and link to the results here when they are available

Persistence model validation runs:
- With all validation samples - [here](https://wandb.ai/manchester_prize/cloudcasting/runs/nnvis171)
- With the validation samples as in this PR - [here](https://wandb.ai/manchester_prize/cloudcasting/runs/yxqpvtik)

And the MAE horizon plots from both are below

W B Chart 19_09_2024, 11_02_05

### CODE REVIEW

File: examples/find_test_2022_t0_times.py [hunk 1]
```diff
@@ -30,17 +33,27 @@
...
 )
 
+# Filter the t0 times so they have gaps of at least 1 hour
+_filtered_t0_times = [available_t0_times[0]]
+
+for t in available_t0_times[1:]:
+    if (t - _filtered_t0_times[-1]) >= MIN_GAP_SIZE:
+        _filtered_t0_times.append(t)
+
+filtered_t0_times = pd.DatetimeIndex(_filtered_t0_times)
```
Comment (phinate) [inline examples/find_test_2022_t0_times.py:51]: this is a fun way of doing this! makes sense :)

File: pyproject.toml [hunk 2]
```diff
@@ -5,7 +5,7 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "cloudcasting"
-version = "0.2.1"
+version = "0.3.0"
```
Comment (phinate) [inline pyproject.toml:8]: fine to leave this version bump in, but this hiding in the diffs has motivated me to see if we can find a better approach!

**Review by phinate — APPROVED** [review]
Plots look good to me. Should help speed things up a lot!

**Review by IFenton — COMMENTED** [review]
LGTM

**Comment by phinate** [comment]
Merged!

**Pull request merged**